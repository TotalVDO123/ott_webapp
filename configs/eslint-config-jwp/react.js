module.exports = {
  extends: [
    // Extend the base config
    './typescript',

    // Use recommended React rules
    'plugin:react/recommended',
  ],

  rules: {
    // Not needed in React 17
    'react/react-in-jsx-scope': 'off',
  },

  overrides: [
    {
      files: ['*.jsx', '*.tsx', '*.ts'],
      plugins: [
        // Enable linting React code
        'react',
        'react-hooks',
      ],
      rules: {
        // Help with Hooks syntax
        'react-hooks/rules-of-hooks': 'error',
        'react-hooks/exhaustive-deps': 'error',

        // Handled by Typescript
        'react/prop-types': 'off',

        // This rule causes too many false positives, e.g. with default exports or child render function
        'react/display-name': 'off',
      },
    },
  ],

  settings: {
    react: {
      pragma: 'React',
      version: '17',
    },
  },
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
  },
};
