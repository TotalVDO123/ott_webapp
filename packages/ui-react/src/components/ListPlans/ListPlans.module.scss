@use '@jwp/ott-ui-react/src/styles/variables';
@use '@jwp/ott-ui-react/src/styles/theme';

.welcome {
  h2 {
    font-weight: var(--body-font-weight-bold);
    font-size: 24px;
  }
}
