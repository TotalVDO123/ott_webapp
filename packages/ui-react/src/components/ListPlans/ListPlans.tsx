import React from 'react';

import Button from '../Button/Button';

import styles from './ListPlans.module.scss';

const ListPlans: React.FC = () => {
  return (
    <div className={styles.welcome}>
      <h2>list plans</h2>
      <div>boxes</div>
      <Button variant="contained" color="primary" label={'Go to Account settings'} onClick={() => {}} size="large" fullWidth />
    </div>
  );
};

export default ListPlans;
