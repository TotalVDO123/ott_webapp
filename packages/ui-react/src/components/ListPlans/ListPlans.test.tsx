import React from 'react';
import { render } from '@testing-library/react';

import ListPlans from './ListPlans';

describe('<ListPlans>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<ListPlans />);

    expect(container).toMatchSnapshot();
  });
});
