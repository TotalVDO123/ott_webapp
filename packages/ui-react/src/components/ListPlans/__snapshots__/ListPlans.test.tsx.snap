// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ListPlans> > renders and matches snapshot 1`] = `
<div>
  <div
    class="_welcome_4e2975"
  >
    <h2>
      list plans
    </h2>
    <div>
      boxes
    </div>
    <button
      class="_button_f8f296 _primary_f8f296 _fullWidth_f8f296 _large_f8f296"
      type="button"
    >
      <span>
        Go to Account settings
      </span>
    </button>
  </div>
</div>
`;
